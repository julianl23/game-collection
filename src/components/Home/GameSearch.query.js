import gql from "graphql-tag";

const GameSearch = gql`
  query GameSearchQuery($query: String!) {
    games(input: { query: $username }) {
      [_id]
    }
  }
`;

export default GameSearch;
