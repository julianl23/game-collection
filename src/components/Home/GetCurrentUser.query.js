import gql from "graphql-tag";

const GET_CURRENT_USER = gql`
  {
    currentUser {
      _id
      firstName
      lastName
      username
      gameCollection {
        items
      }
    }
  }
`;

export default GET_CURRENT_USER;
