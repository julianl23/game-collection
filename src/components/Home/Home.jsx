import React from "react";
import styled from "styled-components";
import TextField from "../TextField";

const HomeSection = styled.section`
  grid-column: 1 / span 4;
`;

const CoverImage = styled.img`
  width: 70px;
  height: 93px;
  background: ${({ theme }) => theme.backgroundGray};
`;

const RecentlyAdded = styled.div`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
`;

const Home = () => (
  <HomeSection>
    <h2>Median</h2>
    <div className="user-counts">
      <p>276 items</p>
      <p>66 wish list</p>
    </div>
    <div className="search">
      <TextField labelText="Search" id="search" />
    </div>
    <RecentlyAdded>
      <CoverImage />
      <CoverImage />
      <CoverImage />
      <CoverImage />
    </RecentlyAdded>
    <div className="friend-activity">
      <div className="feed-item">
        <p>
          VMelo added Injustice: Gods Among Us for the Wii U to their collection
        </p>
      </div>
    </div>
  </HomeSection>
);

export default Home;
