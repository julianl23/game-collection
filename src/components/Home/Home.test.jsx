import React from "react";
import TestRenderer from "react-test-renderer";
import { MockedProvider } from "react-apollo/test-utils";
import wait from "waait";

import Home from "./Home";
import GET_CURRENT_USER from "./GetCurrentUser.query";

const getMockData = ({ currentUser = {} }) => {
  const defaultUser = {
    _id: "123123",
    firstName: "Test",
    lastName: "User",
    username: "testUser",
    gameCollection: {
      items: [],
    },
  };

  return [
    {
      request: {
        query: GET_CURRENT_USER,
      },
      result: {
        data: {
          currentUser: { ...defaultUser, ...currentUser },
        },
      },
    },
  ];
};

describe("Home", () => {
  it("renders correctly", async () => {
    const mocks = getMockData({});
    const instance = TestRenderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Home />
      </MockedProvider>
    );

    await wait(50); // wait for response

    expect(instance.toJSON()).toMatchSnapshot();
  });
});
