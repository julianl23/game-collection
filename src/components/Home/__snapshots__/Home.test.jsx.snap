// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
.c1 {
  margin-bottom: 20px;
}

.c2 {
  display: block;
  margin-bottom: 10px;
}

.c3 {
  display: block;
  border: 1px solid;
  padding: 15px 10px;
  height: 45px;
  width: 100%;
}

.c4 {
  height: 36px;
  background: #322F6F;
  border-radius: 3px;
  font-size: 14px;
  font-weight: 300;
  line-height: 16px;
  color: #fff;
  display: inline-block;
}

.c0 {
  grid-column: 1 / span 4;
}

.c6 {
  width: 70px;
  height: 93px;
}

.c5 {
  display: grid;
  grid-template-columns: repeat(4,1fr);
}

<section
  className="c0"
>
  <h2>
    testUser
  </h2>
  <div
    className="user-counts"
  >
    <p>
      0
       
      items
    </p>
    <p>
      0 wish list
    </p>
  </div>
  <div
    className="search"
  >
    <form
      action="/search"
      method="POST"
      onSubmit={[Function]}
    >
      <div
        className="c1"
      >
        <label
          className="c2"
          htmlFor="query"
        >
          Search
        </label>
        <div
          className="textfield-wrapper"
        >
          <input
            className="c3"
            id="query"
            name=""
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        
      </div>
      <button
        className="c4"
        onClick={null}
        size="normal"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
  <section>
    <h3>
      Recently Added
    </h3>
    <div
      className="c5"
    >
      <img
        className="c6"
      />
      <img
        className="c6"
      />
      <img
        className="c6"
      />
      <img
        className="c6"
      />
    </div>
  </section>
  <div
    className="friend-activity"
  >
    <div
      className="feed-item"
    >
      <p>
        VMelo added Injustice: Gods Among Us for the Wii U to their collection
      </p>
    </div>
  </div>
</section>
`;
